packages/expo-splash-screen/ios/EXSplashScreen/EXSplashScreenService.m

```diff
@@ -0,0 +1,124 @@
+                              successCallback:^{}
+                              failureCallback:^(NSString *message){ UMLogWarn(@"%@", message); }];
+}
+
+- (void)show:(UIViewController *)viewController resizeMode:(EXSplashScreenImageResizeMode)resizeMode
+                                  splashScreenViewProvider:(id<EXSplashScreenViewProvider>)splashScreenViewProvider
+                                           successCallback:(void (^)(void))successCallback
+                                           failureCallback:(void (^)(NSString * _Nonnull))failureCallback
+{
+  if ([_splashScreenControllers objectForKey:viewController]) {
+    return failureCallback(@"'SplashScreen.show' has already been called for given ViewController.");
+  }
+  
+  EXSplashScreenController *splashScreenController = [[EXSplashScreenController alloc] initWithViewController:viewController
+                                                                                                   resizeMode:resizeMode
+                                                                                     splashScreenViewProvider:splashScreenViewProvider];
+  [_splashScreenControllers setObject:splashScreenController forKey:viewController];
+  [[_splashScreenControllers objectForKey:viewController] showWithCallback:successCallback
+                                                           failureCallback:failureCallback];
+}
+
+- (void)preventAutoHide:(UIViewController *)viewController
+        successCallback:(void (^)(void))successCallback
+        failureCallback:(void (^)(NSString * _Nonnull))failureCallback
+{
+  if (![_splashScreenControllers objectForKey:viewController]) {
+    return failureCallback(@"No Native Splash Screen registered for given ViewController. First call 'SplashScreen.show' for given ViewController.");
+  }
+  
+  return [[_splashScreenControllers objectForKey:viewController] preventAutoHideWithCallback:successCallback
+                                                                             failureCallback:failureCallback];
+}
+
+- (void)hide:(UIViewController *)viewController successCallback:(void (^)(void))successCallback
+                                                failureCallback:(void (^)(NSString * _Nonnull))failureCallback
+{
+  if (![_splashScreenControllers objectForKey:viewController]) {
+    return failureCallback(@"No Native Splash Screen registered for given ViewController. First call 'SplashScreen.show' for given ViewController.");
+  }
+  return [[_splashScreenControllers objectForKey:viewController] hideWithCallback:successCallback
+                                                                  failureCallback:failureCallback];
+}
+
+- (void)onAppContentDidAppear:(UIViewController *)viewController
+{
+  if (![_splashScreenControllers objectForKey:viewController]) {
+    UMLogWarn(@"No Native Splash Screen registered for given ViewController. First call 'SplashScreen.show' for given ViewController.");
```

Comment:
In this case both native and splash screen are not proper nouns, so don't capitalize them.
```suggestion
    UMLogWarn(@"No native splash screen registered for given view controller. Call 'SplashScreen.show' for given view controller first.");
```